pkg/core/transactor/transactor.go

```diff
@@ -81,6 +85,11 @@ func (t *Transactor) Listen() {
 	for {
 		select {
 		case r := <-t.stakeChan:
+			// Are we synced?
+			if t.getSyncProgress() != 100 {
+				continue
+			}
```

Comment:
if we are not synced, what happens to the attempted stake transaction? Does it just get garbage collected and forgotten? I don't see a _retry_ mechanism

Reply:
You're absolutely right, forgot to add it. Will do so now